platform/qt/qt.cmake

```diff
@@ -317,18 +317,20 @@ install(TARGETS qmaplibregl
 )
 
 if(NOT MLN_QT_LIBRARY_ONLY)
-    add_executable(
-        mbgl-qt
+    if (COMMAND qt_add_executable)
+        qt_add_executable(mbgl-qt)
+    else()
+        add_executable(mbgl-qt)
+    endif()
+
+    target_sources(mbgl-qt
+        PRIVATE
         ${PROJECT_SOURCE_DIR}/platform/qt/app/main.cpp
```

Comment:
I think you have to store these in a list and then pass them to `add_executable` / `qt_add_executable`/